modules/ROOT/pages/apikit-odatav4-transform-to-sql-operation.adoc

```diff
@@ -0,0 +1,160 @@
+| Entity type keys | Object obtained from the OData request by using #[attributes.odataRequestAttributes], which provides a map of entity key names and values
+| System query options | Parameter obtained from the OData request by using #[attributes.odataRequestAttributes]
+| Search transformation enabled | Enables the $search transformation option for a WHERE clause
+| Strict search term match | Enables full term matching for $search. If enabled, terms in the WHERE clause are not enclosed between '%'.
+| SQL identifiers delimiter a| Option that delimits identifiers in the generated SQL sentence (Options: DOUBLE_QUOTES, BRACKETS, and BACK_TICKS)
+
+Examples:
+
+* DOUBLE_QUOTES: ``SELECT "TrackId", "Name" FROM "Tracks"``
+
+* BRACKETS: ``SELECT [TrackId], [Name] FROM [Tracks]``
+
+* BACK_TICKS: ``SELECT \`TrackId`, \`Name` FROM \`Tracks` ``
+
+| SQL limit rows a| Option that limits the numbers of rows (Options: LIMIT and FETCH_NEXT_ROWS_ONLY)
+
+Examples:
+
+* LIMIT: ``SELECT OrderID, ShippedDate, ShipCountry FROM Orders LIMIT :parameter1``
+
+* FETCH_NEXT_ROWS_ONLY: ``SELECT OrderID, ShippedDate, ShipCountry FROM Orders ORDER BY OrderID ASC OFFSET :parameter2 ROWS FETCH NEXT :parameter1 ROWS ONLY``
+
+`ORDER BY` is automatically included if not present in the request by using the entity key defined in CSDL.
+
+|====
+
+== $filter Expressions in OData V4
+
+​​The OData `$filter` expression uses comparison, logical, arithmetic, and grouping operators. For a list of operators and their meaning, see http://docs.oasis-open.org/odata/odata/v4.01/cs01/part1-protocol/odata-v4.01-cs01-part1-protocol.html#sec_BuiltinFilterOperations[the OData4 Specification^]. 
+
+APIkit does not support the `has` and `divby` operators.
+
+== Expressions Supported by Transform to SQL Select
+
+For the filtering expressions that OData supports see the http://docs.oasis-open.org/odata/odata/v4.01/cs01/part1-protocol/odata-v4.01-cs01-part1-protocol.html#sec_BuiltinFilterOperations[specification^]. 
+
+The `transform-to-sql-select` operation supports the following subset of unary and binary operators, and the following method operators.
+
+=== Binary Operators in OData and in SQL
+
+[options="header"]
+|====
+| OData Operator | SQL Operator
+| add | +
+| and | AND
+| div | /
+| eq | =
+| ge | >=
+| gt | >
+| le | =<
+| lt | <
+| mod | %
+| mul | *
+| ne | <>
+| or | OR
+| sub | -
+|====
+
+=== Unary Operators in OData and in SQL
+
+[options="header"]
+|====
+| OData Operator | SQL Operator
+| not | NOT
+| - | -
+|====
+
+=== Method Operators in OData and in SQL
+
+[options="header"]
+|====
+| OData Operator | SQL Operator
+| contains | LIKE %value%
+| startswith | LIKE value%
+| endswith | LIKE %value
+|====
+
+== $search Support
+
+This operation builds a `where` clause based on the `$search` expression that the system query option provides. The implementation translates the search expression to select all of the elements that have property values that match the condition. 
```

Comment:
We have `where` clauses and WHERE clauses. Are they different? If not, should they be spelled `where` or `WHERE`? Thanks!

Reply:
According to SMEs, both options are possible. But in this case, we need WHERE. Thanks for such a great catch! ;)